packages/nextjs/components/scaffold-eth/Input/InputBase.tsx

```diff
@@ -0,0 +1,50 @@
+import { ChangeEvent, ReactNode, useCallback } from "react";
+import { CommonInputProps } from "./utils";
+
+type InputBaseProps<T> = CommonInputProps<T> & {
+  error?: boolean;
+  disabled?: boolean;
+  prefix?: ReactNode;
+  suffix?: ReactNode;
+};
+
+export const InputBase = <T extends { toString: () => string } = string>({
+  name,
+  value,
+  onChange,
+  placeholder,
+  error,
+  disabled,
+  prefix,
+  suffix,
+}: InputBaseProps<T>) => {
+  const modifiers = [];
+  if (error) {
+    modifiers.push("border-error");
+  } else if (disabled) {
+    modifiers.push("border-disabled bg-base-300");
+  }
+
+  const handleChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      onChange(e.target.value as unknown as T);
+    },
+    [onChange],
+  );
+
+  return (
+    <div className={`flex border-2 border-base-300 bg-base-200 rounded-full text-accent ${modifiers.join(" ")}`}>
```

Comment:
`${modifiers[0]}` ? Or it can be just a string

Reply:
The idea was to allow concurrent modifiers in case we want to expand the input component states (i.e. when `error` and `disabled` are active at the same time). Since it isn't the case yet, I'll simplify it a bit. :+1: